tests/manage/cluster/nodes/base_node_ops.py

```diff
@@ -0,0 +1,104 @@
+import logging
+
+from ocs_ci.framework import config
+from ocs_ci.utility.utils import ceph_health_check
+from ocs_ci.ocs import constants, node
+from ocs_ci.framework.testlib import ManageTest
+from ocs_ci.ocs.cluster import CephCluster
+from tests import helpers
+
+
+logger = logging.getLogger(__name__)
+
+
+class BaseNodes(ManageTest):
+    """
+    Base class for nodes restart related tests
+    """
+    def health_check(self):
+        """
+        Perform Ceph and cluster health checks
+        """
+        ceph_cluster = CephCluster()
+        assert ceph_health_check(
+            namespace=config.ENV_DATA['cluster_namespace']
+        )
+        ceph_cluster.cluster_health_check(timeout=60)
+
+    def validate_cluster(self, resources, instances, health_check=True):
+        """
+        Perform cluster validation - nodes readiness, Ceph cluster health
+        check and functional resources tests
+        """
+        instances_names = list(instances.values())
+        assert node.wait_for_nodes_status(instances_names), (
+            "Not all nodes reached status Ready"
+        )
+        if health_check:
+            self.health_check()
+
+        # Create resources and run IO for both FS and RBD
+        # Unpack resources
+        projects, secrets, pools, storageclasses, pvcs, pods = resources[:6]
+
+        # Project
+        projects.append(helpers.create_project())
+
+        # Secrets
+        secrets.append(helpers.create_secret(constants.CEPHBLOCKPOOL))
+        secrets.append(helpers.create_secret(constants.CEPHFILESYSTEM))
+
+        # Pools
+        pools.append(helpers.create_ceph_block_pool())
+        pools.append(helpers.get_cephfs_data_pool_name())
```

Comment:
that will behave differently as it will return a name and not OCS object

Reply:
below create_storage_class gets pool name so for rbd I'm passing pools[0].name and for fs pools[1]. In the teardown, I need the rbd pool object, pools[0], for deletion